frontend/src/components/Contexts/SafeZoneContext.tsx

```diff
@@ -1,34 +1,58 @@
-import { createContext, FC, useContext, useState } from 'react'
+import { createContext, FC, useContext, useEffect, useState } from 'react'
+import { useRobotContext } from './RobotContext'
+import { useInstallationContext } from './InstallationContext'
+import { AlertType, useAlertContext } from './AlertContext'
+import { SafeZoneBanner } from 'components/Pages/FrontPage/MissionOverview/SafeZoneBanner'
+import { AlertCategory } from 'components/Alerts/AlertsBanner'
 
 interface ISafeZoneContext {
     safeZoneStatus: boolean
-    switchSafeZoneStatus: (newSafeZoneStatus: boolean) => void
 }
 
 interface Props {
     children: React.ReactNode
 }
 
 const defaultSafeZoneInterface = {
-    safeZoneStatus: JSON.parse(localStorage.getItem('safeZoneStatus') ?? 'false'),
-    switchSafeZoneStatus: (newSafeZoneStatus: boolean) => {},
+    safeZoneStatus: false,
 }
 
 export const SafeZoneContext = createContext<ISafeZoneContext>(defaultSafeZoneInterface)
 
 export const SafeZoneProvider: FC<Props> = ({ children }) => {
     const [safeZoneStatus, setSafeZoneStatus] = useState<boolean>(defaultSafeZoneInterface.safeZoneStatus)
-
-    const switchSafeZoneStatus = (newSafeZoneStatus: boolean) => {
-        localStorage.setItem('safeZoneStatus', String(newSafeZoneStatus))
-        setSafeZoneStatus(newSafeZoneStatus)
-    }
+    const { enabledRobots } = useRobotContext()
+    const { installationCode } = useInstallationContext()
+    const { setAlert } = useAlertContext()
+
+    useEffect(() => {
+        const missionQueueFozenStatus = enabledRobots
+            .filter((robot) => robot.currentInstallation.toLocaleLowerCase() === installationCode.toLocaleLowerCase())
+            .map((robot) => robot.missionQueueFrozen)
+            .filter((status) => status === true)
+
+        if (missionQueueFozenStatus.length > 0 && safeZoneStatus === false) {
+            setSafeZoneStatus((oldStatus) => !oldStatus)
+            setAlert(
+                AlertType.RequestSafeZone,
+                <SafeZoneBanner alertCategory={AlertCategory.WARNING} />,
+                AlertCategory.WARNING
+            )
+        } else if (missionQueueFozenStatus.length === 0 && safeZoneStatus === true) {
+            setSafeZoneStatus((oldStatus) => !oldStatus)
+            setAlert(
+                AlertType.DismissSafeZone,
+                <SafeZoneBanner alertCategory={AlertCategory.SUCCESS} />,
+                AlertCategory.SUCCESS
```

Comment:
Existing safe zone alerts should disappear when a new one is added. Can be confusing to see which is relevant. Especially when the banners aren't sorted by time.

Reply:
Have fixed this